client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/task-helper.tsx

```diff
@@ -71,3 +83,58 @@ export function areLaunchpadTasksCompleted(
 
 	return lastTask?.completed;
 }
+
+export function useMaybeSwitchThemeAndUpdateChecklist(
```

Comment:
Maybe the part of the logic that switches the theme should be apart of the plan step exit flow? Doing the switching on the launch means if the user selects a paid theme again it will auto switch again. This assume the user will be able to select paid themes the same way they did before reaching the plan step.

Reply:
I suspect switching when the user makes the decision to use the free plan is the right place for the theme switching to be executed.

That is likely best put inside the submit logic for the modal that pops up on the plan page when you select a free plan, see p1734385074248909-slack-C085HCWCEDN and p1734323301403049-slack-C085HCWCEDN for context.